internal/codeintel/codenav/service.go

```diff
@@ -997,8 +1027,61 @@ func (s *Service) getSyntacticSymbolsAtRange(
 	}
 
 	if parseFail != nil {
-		s.logger.Error("getSyntacticSymbolsAtRange: Failed to parse symbol", log.String("symbol", parseFail.Symbol))
+		s.logger.Warn("getSyntacticSymbolsAtRange: Failed to parse symbol", log.String("symbol", parseFail.Symbol))
 	}
 
 	return symbols, nil
 }
+
+func (s *Service) SyntacticUsages(
+	ctx context.Context,
+	repo types.Repo,
+	commit api.CommitID,
+	path string,
+	symbolRange scip.Range,
+) ([]struct{}, *SyntacticUsagesError) {
+	// The `nil` in the second argument is here, because `With` does not work with custom error types.
+	ctx, trace, endObservation := s.operations.syntacticUsages.With(ctx, nil, observation.Args{Attrs: []attribute.KeyValue{
+		attribute.Int("repoId", int(repo.ID)),
+		attribute.String("commit", string(commit)),
+		attribute.String("path", path),
+		attribute.String("symbolRange", symbolRange.String()),
+	}})
+	defer endObservation(1, observation.Args{})
```

Comment:
I just looked into this; it looks like `endObservation` will log the error if we pass in a proper pointer here. So if we really want to return `SyntacticUsageError` from this function, then we can do the following sequence.

```suggestion
) (_ []struct{}, err *SyntacticUsagesError) {
    var erasedError error
	// The `nil` in the second argument is here, because `With` does not work with custom error types.
	ctx, trace, endObservation := s.operations.syntacticUsages.With(ctx, &erasedError, observation.Args{Attrs: []attribute.KeyValue{
		attribute.Int("repoId", int(repo.ID)),
		attribute.String("commit", string(commit)),
		attribute.String("path", path),
		attribute.String("symbolRange", symbolRange.String()),
	}})
	defer endObservation(1, observation.Args{})
	defer func(){ if err != nil { erasedError = *err } }()
```